### ISSUE TITLE
[OCSP server] handle chain files in -CA

### ISSUE THREAD

**Issue by darkdragon-001** [openssl/openssl#18384]
I am having the same problem as described in [StackOverflow](https://stackoverflow.com/questions/25006998/ocsp-unknown-status-when-passing-cert-good-status-when-passing-serial) already in 2014: When using a chain file for `-CA`, OCSP response is `unknown` instead of `good`. When specifying a file with only the CA certificate (without chain), it returns `good` as intended. I think OpenSSL's OCSP server should be smart enough to just use the first certificate in a chain file.

**Comment by darkdragon-001** [comment]
I was having a brief look at the code and was wondering if we should break the loop after we _found_ the first matching certificate?

https://github.com/openssl/openssl/blob/d1460afdfcb3c90df612896d40d35c6627a5967c/apps/ocsp.c#L1080-L1090

**Comment by darkdragon-001** [comment]
The relevant code is probably:

https://github.com/openssl/openssl/blob/d1460afdfcb3c90df612896d40d35c6627a5967c/apps/ocsp.c#L1080-L1110

For some reason `!found` or `inf == NULL` is true and `UNKNOWN` is returned instead of `GOOD`.

**Comment by darkdragon-001** [comment]
I can't reproduce it any more. I tried the following:
```sh
# Root CA
openssl req -newkey rsa:4096 -noenc -keyout root.key -new -x509 -selfsign -subj "/C=DE/ST=Germany/L=Berlin/O=Testcompany/OU=Testcompany/CN=root/emailAddress=test@testcompany.com" -out root.crt
mkdir root
touch root/index.txt
cat <<-"EOF" > root/openssl.cnf
[ ca ]
default_ca      = CA_default

[ CA_default ]
dir             = root
new_certs_dir   = $dir
database        = $dir/index.txt
serial          = $dir/serial

default_days    = 365

policy          = policy_anything

[ policy_anything ]
countryName             = optional
stateOrProvinceName     = optional
localityName            = optional
organizationName        = optional
organizationalUnitName  = optional
commonName              = supplied
emailAddress            = optional
EOF

# Intermediate CA
openssl req -newkey rsa:4096 -noenc -keyout ca.key -new -sha256 -subj "/C=DE/ST=Germany/L=Berlin/O=Testcompany/OU=Testcompany/CN=ca/emailAddress=test@testcompany.com" -out ca.csr
openssl ca -config root/openssl.cnf -batch -notext -cert root.crt -keyfile root.key -md sha256 -create_serial -in ca.csr -out ca.crt
mkdir ca
touch ca/index.txt
cat <<-"EOF" > ca/openssl.cnf
[ ca ]
default_ca      = CA_default

[ CA_default ]
dir             = ca
new_certs_dir   = $dir
database        = $dir/index.txt
serial          = $dir/serial

default_days    = 365

policy          = policy_anything

[ policy_anything ]
countryName             = optional
stateOrProvinceName     = optional
localityName            = optional
organizationName        = optional
organizationalUnitName  = optional
commonName              = supplied
emailAddress            = optional
EOF

# Client Cert
openssl req -newkey rsa:4096 -noenc -keyout client.key -new -sha256 -subj "/C=DE/ST=Germany/L=Berlin/O=Testcompany/OU=Testcompany/CN=client/emailAddress=test@testcompany.com" -out client.csr
openssl ca -config ca/openssl.cnf -batch -notext -cert ca.crt -keyfile ca.key -md sha256 -create_serial -in client.csr -out client.crt

# OCSP Server
openssl ocsp -index ca/index.txt -CA ca_chain.crt -rsigner ca.crt -rkey ca.key -port 1234 -text
# OCSP Client
openssl ocsp -CAfile root.crt -issuer ca_chain.crt -cert client.crt -text -host localhost:1234
```

Looks like this has been fixed in #941.
